Fixes

owaddell opened the pull request:
- fix typo in mat error messages
- make includes of "unsafe.ss" consistent
- fix a potential crash in `watch-path` callback
- make Swish scripts less broken under Cygwin

payneca committed:
fix typo; do not include the trailing period

owaddell-beckman committed:
fix: avoid potential watch-path crash

libuv now documents that the filename argument to the callback may be
NULL. We have never seen this in practice, but we now check to avoid
crashing in Sstring_utf8 and pass #f to the Scheme callback if filename
is NULL.

Code that handles #(path-changed ,path ,filename ,events) messages
must now be prepared for filename to be #f.

owaddell-beckman committed:
fix: make scripts more viable under Cygwin

Don't let the perfect be the enemy of slightly better.

Swish scripts have been broken under Cygwin, but we can at least try to
make make them work in some typical cases under Cygwin.

- don't rely on OSTYPE; bash doesn't export it by default
- be careful since installing Git Bash changes Cygwin's /tmp
- assume a Cygwin usertemp mount for /tmp
- use registry to get root directory

payneca committed:
fix: use consistent include of unsafe.ss

burgerrg commented on src/swish/app.ss:
```diff
@@ -132,6 +132,35 @@
       (printf "~a~@[ Version ~a~]~@[ (~a)~]\n" name version revision)))
 
+  (meta-cond
+   [(memq (machine-type) '(a6nt ta6nt i3nt ti3nt))
```
`(eqv? (directory-separator) #\\)` will include the ARM versions of Windows in the future should we add them.

owaddell replied:
That's a cute idea. I'm going to try updating the definition in meta.ss and see if we can replace most of the `(machine-type)` calls. Hopefully the build server will let me know if this was misguided.

burgerrg commented on src/swish/app.ss:
```diff
@@ -132,6 +132,35 @@
+        [("/" "tmp" . ,more)
+         (apply path-combine (osi_get_temp_directory) more)]
+        [("/" . ,_)
+         (let ([root (get-registry "hkey_local_machine\\Software\\Cygwin\\setup\\rootdir")])
```
HKEY_LOCAL_MACHINE is typically used in uppercase.

owaddell replied:
Okay, I'll try it in uppercase. The CSUG `get-registry` entry may have misled me.

burgerrg reviewed (approved):
Nice updates! The resolve path was a clever way to handle typical Cygwin paths.

owaddell-beckman committed:
fix: use windows? from (swish meta)

This is easier to maintain and more robust than sprinkling
(machine-type) checks throughout the code.

burgerrg reviewed (approved):
Adding `windows?` worked really well!

owaddell commented:
Thanks for the review and improvements!
